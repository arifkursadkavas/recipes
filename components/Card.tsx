import * as React from 'react';

const Card = (props) => {
  const { title, imageUrl } = props;
  return (
    <div className='cardContainer' onClick={props.onClick}>
      <img
        className='cardImage'
        src={imageUrl}
        alt={title}
        width='100px'
        height='100px'
      ></img>
      <div className='cardTitle'>{title}</div>
    </div>
  );
};

export default Card;
