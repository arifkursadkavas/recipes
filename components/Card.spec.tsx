import * as React from 'react';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Card from './Card';

describe('Components', () => {
  describe('Card', () => {
    test('snapshot renders', () => {
      const component = renderer.create(
        <Card title='Test title' imageUrl='http://image.url.com' />
      );
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});

describe('Components', () => {
  describe('Card', () => {
    it('should render title without throwing an error', function () {
      const wrap = mount(
        <Card title='Test title' imageUrl='http://image.url.com'></Card>
      );
      expect(wrap.find('.cardTitle').text()).toBe('Test title');
    });
  });
});

describe('Components', () => {
  describe('Card', () => {
    it('should render image without throwing an error', function () {
      const wrap = mount(
        <Card title='Test title' imageUrl='http://image.url.com'></Card>
      );
      expect(wrap.find('img').prop('src')).toBe('http://image.url.com');
    });
  });
});

describe('Components', () => {
  describe('Card', () => {
    it('Test Card click event', () => {
      const mockCallBack = jest.fn();

      const card = shallow(<Card onClick={mockCallBack}>Ok!</Card>);
      card.find('.cardContainer').simulate('click');
      expect(mockCallBack.mock.calls.length).toEqual(1);
    });
  });
});
