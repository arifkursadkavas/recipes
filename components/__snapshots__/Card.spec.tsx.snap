// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Card snapshot renders 1`] = `
<div
  className="cardContainer"
>
  <img
    alt="Test title"
    className="cardImage"
    height="100px"
    src="http://image.url.com"
    width="100px"
  />
  <div
    className="cardTitle"
  >
    Test title
  </div>
</div>
`;
